### PR TITLE
Update Truffle, mx, and graal-core versions

### PR DESCRIPTION
**Pull request by mrigger**
Replace methods deprecated in https://github.com/graalvm/truffle/commit/3f99fb6ba44b739109eddd3de388a56098390d4e.

### CODE REVIEW

File: projects/com.oracle.truffle.llvm/src/com/oracle/truffle/llvm/LLVM.java [hunk 1]
```diff
@@ -226,17 +227,21 @@ public static int executeMain(File file, Object... args) {
...
-        Source fromText = Source.fromText(codeString, "code string").withMimeType(LLVMLanguage.LLVM_IR_MIME_TYPE);
-        LLVMLogger.info("current code string: " + codeString);
-        return evaluateFromSource(fromText, args);
+        try {
+            Source fromText = Source.newBuilder(codeString).mimeType(LLVMLanguage.LLVM_IR_MIME_TYPE).build();
+            LLVMLogger.info("current code string: " + codeString);
+            return evaluateFromSource(fromText, args);
+        } catch (MissingNameException e) {
+            throw new AssertionError(e);
+        }
```
Comment (woess) [inline projects/com.oracle.truffle.llvm/src/com/oracle/truffle/llvm/LLVM.java]: you should provide a name, otherwise this will always throw, won't it?

Comment (mrigger) [inline projects/com.oracle.truffle.llvm/src/com/oracle/truffle/llvm/LLVM.java]: Thanks! I did not realize that this method was not used (or just by a single test case that is not part of the CI). I will remove the method altogether.

Comment (woess) [inline projects/com.oracle.truffle.llvm/src/com/oracle/truffle/llvm/LLVM.java]: If it's not tested, it doesn't work. :wink:

Comment (mrigger) [inline projects/com.oracle.truffle.llvm/src/com/oracle/truffle/llvm/LLVM.java]: Yes, I think we need to use coverage tools to identify untested code.

**Pull request merged**